content/blog/2022-04-26-e2e-cv-pt1.md

```diff
@@ -0,0 +1,331 @@
+picture: <TBD>
+pictureComment: <TBD>
+author: alex_kim
+commentsUrl: <TBD>
+tags:
+  - Computer Vision
+  - DVC
+  - CML
+  - Studio
+  - CI/CD
+  - Experiment Tracking
+---
+
+### Introduction
+
+A typical ML project lifecycle goes through three phases:
+
+1. Active exploration or proof-of-concept phase. Here we try many different
+   approaches to data preprocessing, model architectures, and hyperparameter
+   tuning with the goal of finally settling on the most promising combination
+2. Application development phase. Here we build all the “plumbing” around the
+   model, that is: getting input data, massaging it into the right format,
+   passing it to the model, and, finally, serving the model's output. The goal
+   here is to end up with a version of an end-to-end application that is robust
+   and well-performing enough to be considered for deployment to the production
+   environment.
+3. Production deployment phase. Here we promote a well-tested version of the
+   application from the development environment to the production environment.
+
+In this seris of posts we'll describe an approach that streamlines all three
+phases. As our demo project, we've selected a very common deployment pattern in
+CV projects: a CV model wrapped in a web API service. Automatic defect detection
+is an example problem that we are addressing with this pattern. Specifically,
+we'll perform an
+[image segmentation](https://ai.stanford.edu/~syyeung/cvweb/tutorial3.html) task
+on a magnetic tiles dataset first introduced in this
+[paper](https://www.researchgate.net/profile/Congying-Qiu/publication/327701995_Saliency_defect_detection_of_magnetic_tiles/links/5b9fd1bd45851574f7d25019/Saliency-defect-detection-of-magnetic-tiles.pdf)
+and available in this GitHub
+[repository](https://github.com/abin24/Magnetic-tile-defect-datasets.).
+
+![A sample from the image segmentation dataset we used for this project. Top row: images of magnetic tile surfaces. Bottom row: segmentation mask (white pixels show defective areas)](/uploads/images/2022-04-26/dataset_sample.png '=800')
+
+### Target Audience
+
+We assume the target audience of this post to be technical folks (e.g. Software
+Engineers, ML Engineers, Data Scientists) who are familiar with the general
+Machine Learning and Computer Vision concepts, Python programming, CI/CD
+processes, and Cloud infrastructure. Familiarity with the Iterative ecosystem of
+tools such as [DVC](https://dvc.org/), [CML](https://cml.dev/), and
+[Studio](https://studio.iterative.ai/) is not required but would help with
+understanding the nuances of our solution.
+
+### Summary of the Solution
+
+All the code for the project is stored in this GitHub
+[repository](https://github.com/iterative/magnetic-tiles-defect).
+
+The CV API solution that we are proposing can be summarized in the following
+steps:
+
+- Client service will make a POST request with an image as a payload to our API
+  endpoint
```

Comment:
Sorry if this repeats an earlier comment (I can't remember), but I think this sounds overly technical for some of the audience if they are less familiar with software engineering and web apis. Could you simplify to something like "Submit the image to our API endpoint"?